src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -226,6 +226,20 @@ protected override void OnClose(TimeSpan timeout)
             CleanupServerCertificate();
         }
 
+        protected internal override Task OnCloseAsync(TimeSpan timeout)
```

Comment:
Reviewers: this newly added code is to address the issue that some of our types were not overriding these methods and hit the CommunicationObject base class -- which is an error for us.  Please review the code for how I have implemented these methods.  Unfortunately, I am using Task.Run to call synchronous code that has no async equivalent.  Creating async equivalents is a bigger redesign I would prefer not to do here.